app/services/sso_service.rb

```diff
@@ -92,6 +98,7 @@ def invalid_saml_response_handler
     return if saml_response.is_valid?
     fail_handler = SAML::AuthFailHandler.new(saml_response)
     if fail_handler.errors?
+      self.auth_error_code = AUTH_ERRORS[fail_handler.context[:saml_response][:status_message]]
```

Comment:
@saneshark I had to pull the status message from here.. does that seem ok to you?

Reply:
would it be better to set this as attr_reader instead of attr_accessible? Then you would just set an instance variable here. I don't think the caller should ever be in a position to have to set this instance variable.

Using the fail_handler context as you've done here seems fine to me, pending future refactor.